### PR TITLE
Update developer docs for setting up the repo

### LINKED ISSUE #1416: Add documentation around using npm workspaces
Should do the following: vNext
1. remove information about using npm run bootstrap
2. discuss how to manage package-lock.json
3. discuss right way to setup a development environment 

- If using node 14, will need to update version of npm and show expected error if using incorrect version
- npm ci at root level to install packages
- brief description of the node_modules dir (what happens if 2 packages refer to same npm module but at different versions)
- 0.5.0 docs reference vNext docs if using main branch

### PR DESCRIPTION
**Pull request by CaptainIRS**
In this PR:
* The developer docs for setting up the repository are updated with the instructions for utilising npm workspaces

Closes #1416

### CODE REVIEW

File: docs/v0.5.0/CONTRIBUTING.md [hunk 1]
```diff
@@ -45,6 +45,8 @@ The workflow of modifying the Caliper code-base usually consists of the followin
 
 ### Bootstrapping the Caliper repository
 
+> __Note:__ If you are developing from the latest main branch, refer to the instructions in the [vNext contributing guide](https://hyperledger.github.io/caliper/vNext/contributing/). The following instructions apply only apply if you have checked out the v0.5.0 tag and are developing from there.
```
Comment (davidkel) [inline docs/v0.5.0/CONTRIBUTING.md:48]: the link reference ties us to hyperledger-github.io which we don't really want. If you look in https://github.com/hyperledger/caliper/blob/gh-pages/docs/v0.5.0/Fabric_Configuration_New.md you should see examples of how to link to other parts of the documentation.

File: docs/vNext/CONTRIBUTING.md [hunk 2]
```diff
@@ -211,6 +210,46 @@ Each adapter extends the `ConnectorBase` from the core package, as well as expor
...
+npm ERR! notsup Actual:   {"npm":"6.14.17","node":"14.19.3"}
+```
+
+If the npm version is lower than `7.24.2`, you can update it to latest by running the following command:
+
+```console
+npm install -g npm@latest
+```
+
+### Adding new dependencies
```
Comment (davidkel) [inline docs/vNext/CONTRIBUTING.md:237]: Is it similar for updating and uninstalling unused depencencies ?

**Review by davidkel — CHANGES_REQUESTED** [review]
One change required and one question that might be worth addressing in the doc

File: docs/vNext/CONTRIBUTING.md (hunk 2 above)
Comment (CaptainIRS) [reply docs/vNext/CONTRIBUTING.md:237]: Yes. I have now updated the documentation to include uninstall instructions.

**Review by davidkel — APPROVED** [review]
LGTM

**Pull request merged**